pkg/validate/validate_test.go

```diff
@@ -74,63 +74,224 @@ properties:
 `)
 
 func TestValidate(t *testing.T) {
-
 	tests := map[string]struct {
-		f            func(*v1.OpenShiftManagedCluster)
+		f            func(*api.OpenShiftManagedCluster)
 		expectedErrs []error
 		externalOnly bool
 	}{
 		"test yaml parsing": { // test yaml parsing
 
 		},
-		"test version": {
-			f:            func(oc *v1.OpenShiftManagedCluster) { oc.Properties.OpenShiftVersion = "v3.11" },
-			expectedErrs: []error{errors.New("invalid properties.openShiftVersion \"v3.11\"")},
-		},
-		"test Location": {
-			f:            func(oc *v1.OpenShiftManagedCluster) { oc.Location = "" },
+		"location": {
+			f:            func(oc *api.OpenShiftManagedCluster) { oc.Location = "" },
 			expectedErrs: []error{errors.New("invalid location \"\"")},
 		},
-		"test Name": {
-			f:            func(oc *v1.OpenShiftManagedCluster) { oc.Name = "" },
+		"name": {
+			f:            func(oc *api.OpenShiftManagedCluster) { oc.Name = "" },
 			expectedErrs: []error{errors.New("invalid name \"\"")},
 		},
-		"test ProvisioningState": {
-			f:            func(oc *v1.OpenShiftManagedCluster) { oc.Properties.ProvisioningState = "testing" },
-			expectedErrs: []error{errors.New("invalid properties.provisioningState \"testing\"")},
-		},
-		"test master count": {
-			f:            func(oc *v1.OpenShiftManagedCluster) { oc.Properties.MasterPoolProfile.Count = 1 },
-			expectedErrs: []error{errors.New("invalid masterPoolProfile.count 1")},
+		"nil properties": {
+			f:            func(oc *api.OpenShiftManagedCluster) { oc.Properties = nil },
+			expectedErrs: []error{errors.New("properties cannot be nil")},
 		},
 		"test external only true - unset fqdn does not fail": {
-			f:            func(oc *v1.OpenShiftManagedCluster) { oc.Properties.FQDN = "" },
+			f:            func(oc *api.OpenShiftManagedCluster) { oc.Properties.FQDN = "" },
 			externalOnly: true,
 		},
 		"test external only false - unset fqdn fails": {
-			f:            func(oc *v1.OpenShiftManagedCluster) { oc.Properties.FQDN = "" },
+			f:            func(oc *api.OpenShiftManagedCluster) { oc.Properties.FQDN = "" },
 			expectedErrs: []error{errors.New("invalid properties.fqdn \"\"")},
 			externalOnly: false,
 		},
 		"test external only false - invalid fqdn fails": {
-			f:            func(oc *v1.OpenShiftManagedCluster) { oc.Properties.FQDN = "()" },
+			f:            func(oc *api.OpenShiftManagedCluster) { oc.Properties.FQDN = "()" },
 			expectedErrs: []error{errors.New("invalid properties.fqdn \"()\"")},
 			externalOnly: false,
 		},
+		"provisioning state bad": {
+			f:            func(oc *api.OpenShiftManagedCluster) { oc.Properties.ProvisioningState = "bad" },
+			expectedErrs: []error{errors.New("invalid properties.provisioningState \"bad\"")},
```

Comment:
I wish we used backquotes throughout this file, e.g. errors.New(\`invalid properties.provisioningState "bad"\`)

Reply:
wish granted